File: robin/python/baekjoon/brute_force/피로도.py
"""
하루는 24시간.

피로도 -> a
일 - > b
쉬면 -> c
피로도를 최대한 -> m 을 넘지 않게.
"""


def solution():
    a, b, c, m = map(int, input().split())
    time = 24

    tired = 0
    process = 0

    # 피로도가 m 보다 클때
    if a > m:
        print(0)
        return

    while time > 0:

        if tired + a <= m:
            tired += a
            process += b
            time -= 1
        else:
            tired -= c
            time -= 1

    print(process)
    return


if __name__ == '__main__':
    solution()

File: robin/java/src/baekjoon/math/피로도.java
package baekjoon.math;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class 피로도 {

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        String[] input = br.readLine().split(" ");
        // 피로도
        int a = Integer.parseInt(input[0]);

        // 일 처리
        int b = Integer.parseInt(input[1]);

        // 쉴 때
        int c = Integer.parseInt(input[2]);

        // limit
        int m = Integer.parseInt(input[3]);

        int time = 24;
        int tired = 0;
        int process = 0;

        while (time > 0) {


            if (tired + a <= m) {
                tired += a;
                process += b;

            } else {
                tired -= c;

                if (tired < 0) {
                    tired = 0;
                }
            }

            time--;
        }

        System.out.println(process);
    }
}
